network/secondary_endpoint_linux_test.go

```diff
@@ -0,0 +1,284 @@
+			name: "Add endpoints",
+			client: &SecondaryEndpointClient{
+				netlink:        netlink.NewMockNetlink(false, ""),
+				plClient:       platform.NewMockExecClient(false),
+				netUtilsClient: networkutils.NewNetworkUtils(nl, plc),
+				netioshim:      netio.NewMockNetIO(false, 0),
+				ep:             &endpoint{SecondaryInterfaces: make(map[string]*InterfaceInfo)},
+			},
+			epInfo:  &EndpointInfo{MacAddress: mac},
+			wantErr: false,
+		},
+		{
+			name: "Add endpoints invalid mac",
+			client: &SecondaryEndpointClient{
+				netlink:        netlink.NewMockNetlink(false, ""),
+				plClient:       platform.NewMockExecClient(false),
+				netUtilsClient: networkutils.NewNetworkUtils(nl, plc),
+				netioshim:      netio.NewMockNetIO(false, 0),
+				ep:             &endpoint{SecondaryInterfaces: make(map[string]*InterfaceInfo)},
+			},
+			epInfo:     &EndpointInfo{MacAddress: invalidMac},
+			wantErr:    true,
+			wantErrMsg: "SecondaryEndpointClient Error: " + netio.ErrMockNetIOFail.Error() + ": " + invalidMac.String(),
+		},
+		{
+			name: "Add endpoints interface already added",
+			client: &SecondaryEndpointClient{
+				netlink:        netlink.NewMockNetlink(false, ""),
+				plClient:       platform.NewMockExecClient(false),
+				netUtilsClient: networkutils.NewNetworkUtils(nl, plc),
+				netioshim:      netio.NewMockNetIO(false, 0),
+				ep:             &endpoint{SecondaryInterfaces: map[string]*InterfaceInfo{"eth1": {Name: "eth1"}}},
+			},
+			epInfo:     &EndpointInfo{MacAddress: mac},
+			wantErr:    true,
+			wantErrMsg: "SecondaryEndpointClient Error: eth1 already exists",
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			err := tt.client.AddEndpoints(tt.epInfo)
+			if tt.wantErr {
+				require.Error(t, err)
+				require.Contains(t, tt.wantErrMsg, err.Error(), "Expected:%v actual:%v", tt.wantErrMsg, err.Error())
+			} else {
+				require.NoError(t, err)
+				require.Equal(t, tt.client.ep.SecondaryInterfaces["eth1"].MacAddress, tt.epInfo.MacAddress)
+			}
+		})
+	}
+}
+
+func TestSecondaryDeleteEndpoints(t *testing.T) {
+	nl := netlink.NewMockNetlink(false, "")
+	plc := platform.NewMockExecClient(false)
+
+	tests := []struct {
+		name   string
+		client *SecondaryEndpointClient
+		ep     *endpoint
+	}{
+		{
+			name: "Delete endpoint happy path",
+			client: &SecondaryEndpointClient{
+				netlink:        netlink.NewMockNetlink(false, ""),
+				plClient:       platform.NewMockExecClient(false),
+				netUtilsClient: networkutils.NewNetworkUtils(nl, plc),
+				netioshim:      netio.NewMockNetIO(false, 0),
+				nsClient:       NewMockNamespaceClient(),
+			},
+			ep: &endpoint{
+				NetworkNameSpace: "testns",
+				SecondaryInterfaces: map[string]*InterfaceInfo{
+					"eth1": {
+						Name: "eth1",
+						Routes: []RouteInfo{
+							{
+								Dst: net.IPNet{IP: net.ParseIP("192.168.0.4"), Mask: net.CIDRMask(ipv4FullMask, ipv4Bits)},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
```

Comment:
can you add one for failure case where endpoint not removed from map?

Reply:
added new case where we fail to enter namespace